implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -52,10 +52,51 @@ async fn run_impl(
     ctx: Context,
     (opts, cmd): (CommandGlobalOpts, ShowCommand),
 ) -> miette::Result<()> {
-    let node_name = get_node_name(&opts.state, &cmd.node_name);
+    if let Some(node_name) = cmd.node_name {
+        show_node(&opts, &ctx, &node_name).await?;
+    } else if !opts.terminal.can_ask_for_user_input() {
+        show_node(&opts, &ctx, &get_default_node_name(&opts.state)).await?;
+    } else {
+        let node_names = opts.state.nodes.list_items_names()?;
+        if node_names.is_empty() {
+            opts.terminal
+                .stdout()
+                .plain("There are no nodes to show")
+                .write_line()?;
+            return Ok(());
+        }
+
+        let selected_node_names = opts.terminal.select_multiple(
+            "Select one or more nodes that you want to show".to_string(),
+            node_names,
+        );
+
+        if selected_node_names.is_empty() {
+            opts.terminal
+                .stdout()
+                .plain("No nodes selected to show")
+                .write_line()?;
+            return Ok(());
+        }
+
+        if opts.terminal.confirm_interactively(format!(
+            "Would you like to show these items : {:?}?",
+            selected_node_names
```

Comment:
You can use the following notation here to make it less verbose: `format!("Would you like to show these items: {selected_node_names:?}?")`